cmd/minikube/cmd/delete.go

```diff
@@ -725,3 +738,44 @@ func getPids(path string) ([]int, error) {
 
 	return pids, nil
 }
+
+// deleKnownHost read minikube nodes' keys from the minikube home folder
+// and remove them from the known_hosts file
+// This is the long term solution for issue https://github.com/kubernetes/minikube/issues/16868
+func deleteKnownHosts() {
+	// remove this line from known_hosts file if it exists
+	knownHosts := filepath.Join(homedir.HomeDir(), ".ssh", "known_hosts")
+	machinesDir := filepath.Join(localpath.MiniPath(), "machines")
+	fileInfo, err := os.ReadDir(machinesDir)
+	if err != nil {
+		klog.Warningf("error reading machines in minikube home: %v", err)
+		return
+	}
+	if _, err := os.Stat(knownHosts); err == nil {
+		klog.Warningf("no host files found when cleaning host files: %v", err)
+		return
+	}
+	for _, file := range fileInfo {
+		if file.IsDir() {
+			nodeName := file.Name()
+
+			knowHostPath := filepath.Join(localpath.MiniPath(), "machines", nodeName, "known_host")
+			if _, err := os.Stat(knowHostPath); err == nil {
+				// if this file exists, remove this line from known_hosts
+				key, err := os.ReadFile(knowHostPath)
+				if err != nil {
+					klog.Warningf("error reading keys from %s: %v", knowHostPath, err)
+					continue
+				}
+				if err := util.RemoveLineFromFile(string(key), knownHosts); err != nil {
+					klog.Warningf("failed to remove key: %v", err)
+				}
+				// and, remove the file which stores this key
+				if err := os.Remove(knowHostPath); err != nil {
+					klog.Warningf("failed to remove key: %v", err)
+				}
+			}
+		}
```

Comment:
```suggestion
		if !file.IsDir() {
			continue
		}
		nodeName := file.Name()

		knowHostPath := filepath.Join(localpath.MiniPath(), "machines", nodeName, "known_host")
		if _, err := os.Stat(knowHostPath); err != nil {
			continue
		}
		// if this file exists, remove this line from known_hosts
		key, err := os.ReadFile(knowHostPath)
		if err != nil {
			klog.Warningf("error reading keys from %s: %v", knowHostPath, err)
			continue
		}
		if err := util.RemoveLineFromFile(string(key), knownHosts); err != nil {
			klog.Warningf("failed to remove key: %v", err)
		}
		// and, remove the file which stores this key
		if err := os.Remove(knowHostPath); err != nil {
			klog.Warningf("failed to remove key: %v", err)
		}
```
Invert `if`s to reduce deep nesting